arbos/arbostypes/incomingmessage.go

```diff
@@ -34,6 +35,20 @@ const (
 
 const MaxL2MessageSize = 256 * 1024
 
+type BlockMerkleJustification struct {
+	// Merkle proof validating the justification header.
+	BlockMerkleProof *espressoTypes.HotShotBlockMerkleProof
+	// Block merkle commitment that corresponds to a valid light client snapshot.
+	BlockMerkleComm *espressoTypes.TaggedBase64
+}
+
+type EspressoBlockJustification struct {
```

Comment:
The Justification is used to justify that the L2 blocks are created from the transactions sequenced by the Espresso sequencer, or in other words went through the HotShot consensus.